src/stream/reader/logfile.rs

```diff
@@ -0,0 +1,373 @@
+                // We are using seek_relative which makes things slightly
+                // more complicated, but it does drop caches if it is not necessary
+                // in comparison to self.file.seek
+                let diff = chunk.position as i64 - last_pos;
+                self.file.seek_relative(diff)?;
+                buffer.resize(chunk.size as usize, 0u8);
+                self.file.read_exact(&mut buffer)?;
+                lock.write_all(&buffer)?;
+                last_pos = chunk.position as i64 + chunk.size as i64;
+            }
+        }
+        Ok(())
+    }
+
+    fn read_block(file: &mut BufReader<File>) -> anyhow::Result<Option<Block>> {
+        match file.read_u8() {
+            Ok(BLOCK_STREAM_START) => {
+                let task_id = file.read_u32::<byteorder::BigEndian>()?;
+                let instance_id = file.read_u32::<byteorder::BigEndian>()?;
+                Ok(Some(Block::StreamStart {
+                    task_id,
+                    instance_id,
+                }))
+            }
+            Ok(BLOCK_STREAM_CHUNK) => {
+                // Job task stream data
+                let task_id = file.read_u32::<byteorder::BigEndian>()?;
+                let instance_id = file.read_u32::<byteorder::BigEndian>()?;
+                let channel_id = file.read_u32::<byteorder::BigEndian>()?;
+                let size = file.read_u32::<byteorder::BigEndian>()?;
+                Ok(Some(Block::StreamChunk {
+                    task_id,
+                    instance_id,
+                    channel_id,
+                    size,
+                }))
+            }
+            Ok(BLOCK_STREAM_END) => {
+                let task_id = file.read_u32::<byteorder::BigEndian>()?;
+                let instance_id = file.read_u32::<byteorder::BigEndian>()?;
+                Ok(Some(Block::StreamEnd {
+                    task_id,
+                    instance_id,
+                }))
+            }
+            Err(e) => match e.kind() {
+                ErrorKind::UnexpectedEof => Ok(None),
+                _ => Err(e.into()),
+            },
+            Ok(r) => anyhow::bail!("Invalid  entry type: {}", r),
+        }
+    }
+
+    pub fn show(&mut self, opts: &ShowOpts) -> anyhow::Result<()> {
+        let id_width = if let Some(max_id) = self.index.keys().max() {
+            max_id.to_string().len()
+        } else {
+            return Ok(());
+        };
+        self.file.seek(SeekFrom::Start(self.start_pos))?;
+        let mut buffer = Vec::new();
+
+        let colors = [
+            Color::Red,
+            Color::Green,
+            Color::Yellow,
+            Color::Blue,
+            Color::Magenta,
+            Color::Cyan,
+        ];
+
+        let selected_channel_id = opts.channel.as_ref().map(|c| match c {
+            Channel::Stdout => 0,
+            Channel::Stderr => 1,
+        });
+
+        let stdout = std::io::stdout();
+        let mut _lock = stdout.lock();
```

Comment:
This `_lock` should be used instead of `println`.

Reply:
I think that the lock is reentrant and locking an already locked lock is fast, but yes, calling directly method on lock is probably slightly faster

Reply:
Yes, it is reentrant. But it looks really weird to lock it and then call `println!`, I originally expected that there will be a deadlock :)

Reply:
It is also in perf book: https://nnethercote.github.io/perf-book/io.html :)

Reply:
Ok, not. I sould read more carefully :)